log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -144,13 +152,17 @@ private void bufferDrainLoop()
                             // Note: queuePollFirst() will return null and set terminalMessageDequeued when encountering TERMINAL_MESSAGE.
                             message = queuePollFirst();
 
-                            if (!dropSnapshot.isEmpty() || message != null) {
+                            flushedSignal = this.flushedSignal;
+                            this.flushedSignal = null;
```

Comment:
add a comment saying that this code will handle the flush signal, so move it to a local variable.